cpp/src/arrow/buffer.h

```diff
@@ -402,6 +402,17 @@ Status AllocateEmptyBitmap(MemoryPool* pool, int64_t length,
 ARROW_EXPORT
 Status AllocateEmptyBitmap(int64_t length, std::shared_ptr<Buffer>* out);
 
+/// \brief Concatenate multiple buffers into a single buffer
+///
+/// \param[in] buffers to be concatenated
+/// \param[in] pool memory pool to allocate the new buffer from
+/// \param[out] out the concatenated buffer
+///
+/// \return Status
+ARROW_EXPORT
+Status Concatenate(const std::vector<std::shared_ptr<Buffer>>& buffers, MemoryPool* pool,
```

Comment:
the renaming seems unnecessary- the overload will not conflict with concatenate(arrays)

Reply:
I'm not concerned about conflict. I'm concerned about readability. `buffer::Concatenate` or `Buffer::Concatenate` speaks a lot more than just `Concatenate`.

Reply:
IMHO it's more readable to avoid manual name mangling; the argument types are part of what users will read

Reply:
Would it help to introduce an alias like:
```
using BufferVector = std::vector<std::shared_ptr<Buffer>>
``` 
I think this would help make the signature easier to read.